R/package-installation-methods.R

```diff
@@ -166,6 +172,61 @@ versioned_install_instructions <- function(pkgs) {
   return(instructions)
 }
 
+deps_table = function(packages) {
+  res = sessioninfo::package_info(pkgs = packages)
+  res = as.data.frame(res)[, c("package", "ondiskversion")]
+  colnames(res) = c("package", "version")
+  res$type = "Imports"
+  na_version = is.na(res$version)
+  res$version[!na_version] = paste0(">= ", res$version[!na_version])
+  res$version[na_version] = "*"
+  res = res[, c("type", "package", "version")]
+  res
+}
+
+
+fake_description_from_packages = function(packages) {
+  deps = deps_table(packages)
+  desc = desc::description$new("!new")
+  desc$set_deps(deps)
+  desc$set("Package", "base")
+  tfile = tempfile()
+  desc$write(tfile)
+  tfile
+}
+
+
+.find_sys_deps = function(packages,
+                          platform,
+                          soft = TRUE,
+                          offline = FALSE) {
+  stopifnot(is.logical(offline) && length(offline) == 1)
+  method = ifelse(offline, "sysreq-package", "sysreq-api")
```

Comment:
Note to self: this is a good spot where to check if `sysreqs` is available, and proceed differently but gracefully if not. Important for #68